tests/time_tests/scripts/compare_runs.py

```diff
@@ -0,0 +1,205 @@
+    ])
+    # `Comparison_ops` is a template applied for metrics columns
+    # generated by applied `ops` to propagate status of function
+    comparison_ops = {
+        # format: {metric_col_name: (operation, message)}
+        metric_name_template.format(required_metrics[0], "ref/cur"):
+            lambda x: (gmean(x) > GEOMEAN_THRESHOLD,
+                       "geomean={} is less than threshold={}".format(gmean(x), GEOMEAN_THRESHOLD)),
+    }
+
+    filtered_refs = {}
+    for record in references:
+        filtered_rec = {key: val for key, val in record.items() if key in required_fields}
+        filtered_rec_metrics = {key: val for key, val in record["results"].items() if key in required_metrics}
+        filtered_refs[HashableDict(filtered_rec)] = filtered_rec_metrics
+
+    filtered_cur_val = {}
+    for record in cur_values:
+        filtered_rec = {key: val for key, val in record.items() if key in required_fields}
+        filtered_rec_metrics = {key: val for key, val in record["results"].items() if key in required_metrics}
+        filtered_cur_val[HashableDict(filtered_rec)] = filtered_rec_metrics
+
+    comparison_data = []
+    for data in [filtered_refs, filtered_cur_val]:
+        for record in data:
+            rec = OrderedDict()
+            for field in required_fields:
+                rec.update({field: record[field]})
+                rec.move_to_end(field)
+            if rec not in comparison_data:
+                # Comparison data should contain unique records combined from references and current values
+                comparison_data.append(rec)
+    comparison_data = sorted(comparison_data, key=itemgetter("model_name"))
+
+    for record in comparison_data:
+        metrics_rec = OrderedDict()
+        for metric in required_metrics:
+            ref = filtered_refs.get(HashableDict(record), {}).get(metric, None)
+            cur = filtered_cur_val.get(HashableDict(record), {}).get(metric, None)
+            for op_name, op in ops.items():
+                op_res = op(ref, cur)
+                metric_name = metric_name_template.format(metric, op_name)
+                metrics_rec.update({metric_name: op_res})
+                metrics_rec.move_to_end(metric_name)
+        # update `comparison_data` with metrics
+        for metric_name, op_res in metrics_rec.items():
+            record.update({metric_name: op_res})
+            record.move_to_end(metric_name)
+
+    # compare data using `comparison_ops`
+    orig_data = pandas.DataFrame(comparison_data)
+    data = orig_data.dropna()
+    
+    devices = data["device"].unique()
+    for device in devices:
+        frame = data[data["device"] == device]
+        for field, comp_op in comparison_ops.items():
+            status, msg = comp_op(frame.loc[:, field])
+            if not status:
+                log.error('Comparison for field="%s" for device="%s" failed: %s', field, device, msg)
```

Comment:
add information about comparison status and metrics in report (HTML or any other report)